### PR TITLE
[paddle] add documentation

### PR DESCRIPTION
**Pull request by ksivaman**

### CODE REVIEW

File: docs/api/paddle.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    See LICENSE for license information.
+
+paddle
+======
+
+.. autoapiclass:: transformer_engine.paddle.Linear(in_features, out_features, **kwargs)
+  :members: forward
+
+.. autoapiclass:: transformer_engine.paddle.LayerNorm(hidden_size, eps=1e-5, **kwargs)
```
Comment (timmoon10) [inline docs/api/paddle.rst:12]: For consistency:
```suggestion
.. autoapiclass:: transformer_engine.paddle.LayerNorm(hidden_size, eps=1e-5, **kwargs)
  :members: forward
```

File: transformer_engine/paddle/layer/attention.py [hunk 2]
```diff
@@ -387,7 +391,8 @@ class MultiHeadAttention(paddle.nn.Layer):
     zero_centered_gamma: bool, default = `False`
                     whether to zero initialize the gamma of the layernorm operation.
     backend: {'transformer_engine', 'paddle'}, default = `transformer_engine`
-                backend to use for attention operation.
+             backend to use for attention operation. if set to 'paddle', a framework
```
Comment (timmoon10) [inline transformer_engine/paddle/layer/attention.py:394]:
```suggestion
             backend to use for attention operation. If set to 'paddle', a framework
```

File: transformer_engine/paddle/layer/layernorm.py [hunk 3]
```diff
@@ -63,7 +63,33 @@ def backward(ctx, grad_output: paddle.Tensor) -> Tuple[Union[paddle.Tensor, None
...
+              optional `paddle.ParamAttr` for bias.
+    zero_centered_gamma : bool, default = 'False'
+                         if set to 'True', gamma parameter in LayerNorm is initialized to 0 and
+                         the LayerNorm formula changes to
+
+                         .. math::
+                            y = \frac{x - \mathrm{E}[x]}{ \sqrt{\mathrm{Var}[x] + \varepsilon}} *
+                            (1 + \gamma) + \beta
+    backend: {'transformer_engine', 'paddle'}, default = `transformer_engine`
+             backend to use for softmax operation.
```
Comment (timmoon10) [inline transformer_engine/paddle/layer/layernorm.py:92]:
```suggestion
             backend to use for operation.
```

**Review by timmoon10 — APPROVED** [review]
Overall LGTM

File: transformer_engine/paddle/layer/layernorm.py (hunk 3 above)
Comment (ksivaman) [reply transformer_engine/paddle/layer/layernorm.py:92]: Done

File: transformer_engine/paddle/layer/attention.py (hunk 2 above)
Comment (ksivaman) [reply transformer_engine/paddle/layer/attention.py]: Done

File: docs/api/paddle.rst (hunk 1 above)
Comment (ksivaman) [reply docs/api/paddle.rst:12]: We don't have this for other frameworks, so just keeping it consistent

**Pull request merged**